build/index_core_compatible.js

```diff
@@ -1,20 +1,37 @@
-import Experiment from '../es6/experiment';
-import Interpreter from '../es6/interpreter';
-import RandomPlanoutCoreCompatible from '../es6/ops/randomPlanoutCoreCompatible';
-import Core from '../es6/ops/core';
-import * as Namespace from '../es6/namespace';
 import Assignment from '../es6/assignment';
+import Experiment from '../es6/experiment';
 import ExperimentSetup from '../es6/experimentSetup';
+import Interpreter from '../es6/interpreter';
+import * as LibUtils from '../es6/lib/utils';
+import Random from '../es6/ops/randomPlanoutCoreCompatible';
+import * as Base from '../es6/ops/base';
+import * as Core from '../es6/ops/core';
+import * as OpsUtils from '../es6/ops/utils';
+import * as Namespace from '../es6/namespacePlanoutCoreCompatible';
+
+OpsUtils.registerOperators({
+    "randomFloat": Random.RandomFloat,
+    "randomInteger": Random.RandomInteger,
+    "bernoulliTrial": Random.BernoulliTrial,
+    "bernoulliFilter": Random.BernoulliFilter,
+    "uniformChoice": Random.UniformChoice,
+    "weightedChoice": Random.WeightedChoice,
+    "sample": Random.Sample
+}, true);
```

Comment:
Unfortunately I think we're wrestling a bit with the current structure of the project to force it to work the way that we want. In an ideal world, we shouldn't have to apply any "overrides" to our random operations at runtime.

The direction that I think PlanOut.js should move in is one where we totally separate the concerns of API & random operations. PlanOut.js's experiments, namespaces, interpreter, etc should be capable of being composed from the top with whatever random operations we choose. Right now we're in an awkward middle ground where some stuff is composable, but other stuff directly imports random operations.

Here's a peak into what I'm imagining:

- /
  - /build
    - index.js
    - index_core_compatible.js
  - /es6
    - /api
      - ...
    - /ops
      - ...

I'm not married to this exact directory structure, but the important detail is that we should have a hard rule that no module in ```/api``` will ever import directly from ```/ops```. Anything that an api module needs from ```/ops``` should be passed down from the build's entry point.

With this restructuring, I think that PlanOut.js will be in a good place to scale without having to worry about future changes causing a shift in enrollment for either the compat or non-compat builds.

Taking a step back to the problem at hand...

The fact is that things are broken, and they need to be fixed. The question I'm interested in answering right now is - _do we apply a bandaid to fix core-compat, or do we prioritize a more involved project restructure?_

@jcwilson do you have any thoughts on the matter? How high priority is fixing core-compat to you and your team?

@rawls238 do you think that what I'm proposing sounds like a reasonable direction to head in?

/cc @geoffdaigle